### ISSUE TITLE
Stratum Connection Failed but Still Good on rbminer.net

### ISSUE THREAD

**Issue by jch9678** [RainbowMiner/RainbowMiner#291]
I'm getting a stratum connection failed with z-enemy for hex at nlpool.  rbminer is showing that everything is ok.  It's probably showing that everything is ok because RainbowMiner is not switching.  It has been getting this failure for almost 30 minutes.  Seems like it should've switched by now.

**Comment by hernandn01** [comment]
same happened to me, I disabled hex on nlpool

**Comment by jch9678** [comment]
Yeah, that's a good quick fix but the real problem is why RB never switched...

**Comment by RainbowMiner** [comment]
Probably, because z-enemy continues to show hashrate numbers over the API. I will try to find a way to work around this.

**Comment by RainbowMiner** [comment]
Would you please go to http://localhost:4000 and click "Debug file" at the left bar and drop the created 7z file here? The file contains all current logs plus config with all critical values x-ed out, so no wallet addresses, api keys, usernames will be exposed.

**Comment by jch9678** [comment]
github doesn't support 7z.  I'll create a zip file.

**Comment by jch9678** [comment]
[debug-2018-11-08.zip](https://github.com/RainbowMiner/RainbowMiner/files/2562384/debug-2018-11-08.zip)

**Comment by jch9678** [comment]
Would you mind taking a quick look at this debug file from a different machine.  It keeps crashing about every 8 or so hours.
[debug-2018-11-08.zip](https://github.com/RainbowMiner/RainbowMiner/files/2562443/debug-2018-11-08.zip)

**Comment by RainbowMiner** [comment]
Thank you :) Yes, I will have a look. Sorry for 7z archives. I had forgotten that github likes zip only. I already changed the zipping for the next release.

**Comment by RainbowMiner** [comment]
> Would you mind taking a quick look at this debug file from a different machine. It keeps crashing about every 8 or so hours.

The log has been too short to evaluate properly, so I changed the Debug file creation for the next version. Anyway: I suspect SrbMiner to crash the machine when closing down the miner. I would check the config files, if the one or the other has problematic settings (maybe reduce a bit).

**Comment by RainbowMiner** [comment]
Yes, the key is the new API.psm1 file, found here: https://raw.githubusercontent.com/RainbowMiner/RainbowMiner/master/API.psm1 (you could download it with right-click, right into your Rainbowminer folder).
After that, the logs will contain two days, and be in zip format.

**Comment by hernandn01** [comment]
not sure if the new API.psm1 will fix the original question. I got this morning again a fail connection from Hex on NLPool. RBMiner.net was reporting 0 hashrate but it wasn't disconnecting until I disable hex on Nlpool

**Comment by jch9678** [comment]
It won't fix that problem, it just allows him to get easy access to more debugging files.  We still have 2 problems:
1) need a workaround for enemy zealot reporting a hashrate when it has failed
2) why is enemy zealot failing, maybe it has the wrong port.

**Comment by jch9678** [comment]
2) It's on the right port 5135, 
A solution to 1) could be that RainbowMiner switches miners after a reading of 0 being outside the fault tolerance for some time, maybe 2 minutes...

**Comment by hernandn01** [comment]
checked the port for hex at Nlpool and it was using the correct one

2018-11-09 05:13:03 [531.79 483.50] INFO: OC set for EnemyZealot-GTX10603GB-Hex: PL=75%, TL=-, MEM=0, CORE=100, LVP=850000µV
2018-11-09 05:13:04 [531.79 483.50] INFO: Starting miner (EnemyZealot-GPU#04): 'C:\miner\RainbowMiner\Bin\NVIDIA-enemyz\z-enemy.exe -R 1 -b 30204 -d 3 -a hex -q -o stratum+tcp://mine.nlpool.nl:**5135** -u 3CnnvybE8YvhrZeqW3Rw6tD7XZ1Rk3dXye -p worker1,c=BTC -N 1'
2018-11-09 05:13:06 [541.43 493.22] INFO: Start waiting 136 seconds before next run. 
2018-11-09 05:15:23 [433.81 487.83] INFO: Pinging monitoring server. 
2018-11-09 05:15:24 [433.82 487.84] INFO: Miner Status https://rbminer.net/api/report.php: Success
2018-11-09 05:15:24 [433.82 487.84] INFO: Finish waiting before next run. 
2018-11-09 05:15:24 [433.82 487.84] INFO: Saving hash rates. 
2018-11-09 05:15:24 [433.90 487.84] WARNING: Stat file (EnemyZealot-GPU#02_Hex_HashRate) was not updated because the value **(0)** is outside fault tolerance (3405972 to 10899109). 
2018-11-09 05:15:24 [433.90 487.84] WARNING: Stat file (EnemyZealot-GPU#04_Hex_HashRate) was not updated because the value **(0)** is outside fault tolerance (3257041 to 10422530). 

![image](https://user-images.githubusercontent.com/41829368/48273074-59f10980-e3fd-11e8-820b-b6c93dd1e8d6.png)

**Comment by jch9678** [comment]
In the meantime I've disabled NLpool...

**Comment by hernandn01** [comment]
it is not a miner issue. Zealot mining Hex at Blockmasters

![image](https://user-images.githubusercontent.com/41829368/48273490-498d5e80-e3fe-11e8-9605-640ab93561ed.png)

**Comment by RainbowMiner** [comment]
Yes, the API.psm1 is just to be able to create a debug log for two days. It is not a fix for the other problem.
RainbowMiner includes a watchdog which -should- put brakes on such misbehaving miners. I am still analysing why this is not working for Enemy/NLpool combination.
I will soon push a new release, including a possible fix for the problem.
